jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/TestHttpBuildPacksArtifactsServer.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.common;
+
+import org.apache.commons.io.FileUtils;
+import org.eclipse.jkube.kit.common.util.FileUtil;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+public class TestHttpBuildPacksArtifactsServer implements Closeable {
+  private final TestHttpStaticServer testHttpStaticServer;
+  private static final String LINUX_ARTIFACT = "pack-v0.32.1-linux.tgz";
+  private static final String LINUX_ARM64_ARTIFACT = "pack-v0.32.1-linux-arm64.tgz";
+  private static final String MACOS_ARTIFACT = "pack-v0.32.1-macos.tgz";
+  private static final String MACOS_ARM64_ARTIFACT = "pack-v0.32.1-macos-arm64.tgz";
+  private static final String WINDOWS_ARTIFACT = "pack-v0.32.1-windows.zip";
+  private final File remoteBuildPackArtifactsDir;
+
+  public TestHttpBuildPacksArtifactsServer() throws IOException {
```

Comment:
Since this is a test util, maybe we want to skip the checked exception altogether and just throw an IllegalStateException